salesforce/utils.bal

```diff
@@ -13,67 +13,17 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
+
 import ballerina/log;
 import ballerina/http;
-import ballerina/url;
-
-# Returns the prepared URL.
-#
-# + paths - An array of paths prefixes
-# + return - The prepared URL
-public isolated function prepareUrl(string[] paths) returns string {
-    string url = EMPTY_STRING;
-
-    if paths.length() > 0 {
-        foreach string path in paths {
-            if !path.startsWith(FORWARD_SLASH) {
-                url = url + FORWARD_SLASH;
-            }
-            url = url + path;
-        }
-    }
-    return url;
-}
-
-# Returns the prepared URL with encoded query.
-#
-# + paths - An array of paths prefixes
-# + queryParamNames - An array of query param names
-# + queryParamValues - An array of query param values
-# + return - The prepared URL with encoded query
-isolated function prepareQueryUrl(string[] paths, string[] queryParamNames, string[] queryParamValues) returns string {
-    string url = prepareUrl(paths);
-
-    url = url + QUESTION_MARK;
-    boolean first = true;
-    int i = 0;
-    foreach string name in queryParamNames {
-        string value = queryParamValues[i];
-        string|url:Error encoded = url:encode(value, ENCODING_CHARSET);
-
-        if encoded is string {
-            if first {
-                url = url + name + EQUAL_SIGN + encoded;
-                first = false;
-            } else {
-                url = url + AMPERSAND + name + EQUAL_SIGN + encoded;
-            }
-        } else {
-            log:printError("Unable to encode value: " + value, 'error = encoded);
-            break;
-        }
-        i = i + 1;
-    }
-    return url;
-}
 
 # Check HTTP response and return JSON payload if succesful, else set errors and return Error.
 #
 # + httpResponse - HTTP respone or Error
 # + expectPayload - Payload is expected or not
 # + return - JSON result if successful, else Error occured
-isolated function checkAndSetErrors(http:Response|error httpResponse, boolean expectPayload = true)
-                                    returns json|Error {
+isolated function checkAndSetErrors(http:Response|error httpResponse, boolean expectPayload = true) 
+                                    returns @tainted json|Error {
```

Comment:
Why are we adding `tainted`?

Reply:
Removed as it is unnecessary in fc08a84ae3b62f98c8863f7b4b430da95533b776